test-d/writable-deep.ts

```diff
@@ -64,7 +64,8 @@ const data = {
 const readonlyData: ReadonlyDeep<typeof data> = data;
 
 let writableData: WritableDeep<typeof readonlyData>;
-expectError(writableData = readonlyData);
+// @ts-expect-error
+writableData = readonlyData; // eslint-disable-line prefer-const
```

Comment:
We can't, `@ts-expect-error` is conflict with `ExpectError`, we only can use one of them . As for why, I haven't looked into him deeply enough.

Reply:
> As for why, I haven't looked into him deeply enough.

`expectError()` checks if the compiler did raise an error, but `// @ts-expect-error` instructs the compiler to ignore the error. If the error is not raised because of `// @ts-expect-error`, `expectError()` cannot perform the check.

Other notable difference. `expectError()` is silencing only a predefined list of semantic (or type) errors. In contrary `// @ts-expect-error` is silencing all and every error including: missing imports, typos, syntax errors etc.

Reply:
> missing imports, typos, syntax errors etc.

This is indeed a problem, and I've tried to manually eliminate as many of these extra factors as possible, but this is not a complete solution.